src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotator.java

```diff
@@ -0,0 +1,293 @@
+    public double minGenotypeQualityP = 0.0;
+
+    private VariantAnnotatorEngine annotatorEngine;
+    private SampleList variantSamples;
+
+    /**
+     * Prepare the output file and the list of available features.
+     */
+    public void onTraversalStart() {
+        // get the list of all sample names from the variant VCF input rod, if applicable
+        final  List<String> samples = getHeaderForVariants().getGenotypeSamples();
+        variantSamples = new IndexedSampleList(samples);
+
+        if ( USE_ALL_ANNOTATIONS ) {
+            annotatorEngine = VariantAnnotatorEngine.ofAllMinusExcluded(variantAnnotationArgumentCollection.annotationsToExclude, dbsnp.dbsnp, comps, false);
+        } else {
+            annotatorEngine = VariantAnnotatorEngine.ofSelectedMinusExcluded(variantAnnotationArgumentCollection, dbsnp.dbsnp, comps, false);
+        }
+        annotatorEngine.addExpressions(expressionsToUse, resources, expressionAlleleConcordance );
+
+        // setup the header fields
+        // note that if any of the definitions conflict with our new ones, then we want to overwrite the old ones
+        final Set<VCFHeaderLine> hInfo = new HashSet<>();
+
+        hInfo.addAll(annotatorEngine.getVCFAnnotationDescriptions(false));
+        hInfo.addAll(getHeaderForVariants().getMetaDataInInputOrder());
+
+        // for the expressions, pull the info header line from the header of the resource rod
+        for ( final VariantAnnotatorEngine.VAExpression expression : annotatorEngine.getRequestedExpressions() ) {
+            // special case the ID field
+            if ( expression.fieldName.equals("ID") ) {
+                hInfo.add(new VCFInfoHeaderLine(expression.fullName, 1, VCFHeaderLineType.String, "ID field transferred from external VCF resource"));
+            } else {
+                final VCFInfoHeaderLine targetHeaderLine = ((VCFHeader) getHeaderForFeatures(expression.binding)).getInfoHeaderLines().stream()
+                            .filter(l -> l.getID().equals(expression.fieldName))
+                            .findFirst().orElse(null);
+
+                VCFInfoHeaderLine lineToAdd;
+                if (targetHeaderLine != null) {
+                    expression.hInfo = targetHeaderLine;
+                    if (targetHeaderLine.getCountType() == VCFHeaderLineCount.INTEGER) {
+                        lineToAdd = new VCFInfoHeaderLine(expression.fullName, targetHeaderLine.getCount(), targetHeaderLine.getType(), targetHeaderLine.getDescription());
+                    } else {
+                        lineToAdd = new VCFInfoHeaderLine(expression.fullName, targetHeaderLine.getCountType(), targetHeaderLine.getType(), targetHeaderLine.getDescription());
+                    }
+                } else {
+                    lineToAdd = new VCFInfoHeaderLine(expression.fullName, VCFHeaderLineCount.UNBOUNDED, VCFHeaderLineType.String, "Value transferred from another external VCF resource");
```

Comment:
I don't think there's anything better to do here, but I can imagine a scenario where someone forgets some annotation, adds it to a GVCF, tries to import it into GenomicsDB and the yogurt hits the fan.  (You'd be surprised what users will do when left to their own devices.)  Maybe we can output a warning?  In theory this has the potential to break BCF output in any form.

Reply:
I added a warning, it does seem like something must have gone wrong if the header line is missing.